community/logging/src/main/java/org/neo4j/logging/FormattedLog.java

```diff
@@ -203,19 +207,30 @@ protected void writeLog( PrintWriter out, String message )
         @Override
         protected void writeLog( PrintWriter out, String message, Throwable throwable )
         {
+            requireNonNull( message, "message" );
+
             lineStart( out );
             out.write( message );
-            out.write( ' ' );
-            out.write( throwable.getMessage() );
-            out.println();
-            throwable.printStackTrace( out );
+            if ( throwable != null )
+            {
+                if ( throwable.getMessage() != null )
+                {
+                    out.write( ' ' );
+                    out.write( throwable.getMessage() );
+                }
+                out.println();
+                throwable.printStackTrace( out );
+            }
         }
 
         @Override
         protected void writeLog( PrintWriter out, String format, Object[] arguments )
         {
+            requireNonNull( format, "format" );
+
+            String message = String.format( format, arguments );
```

Comment:
I guess so. Feels wasteful to build a whole string in memory, just to then write it out - when it could be written out directly. I wonder if there is a way to validate without doing the actual message construction?